source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp

```diff
@@ -1939,6 +1939,9 @@ unsigned SwiftExpressionParser::Parse(DiagnosticManager &diagnostic_manager,
   //  - Some passes may create new functions, but only the functions defined in
   //    the lldb repl line should be serialized.
   if (swift_ast_ctx->UseSerialization()) {
+    // Run all the passes before differentiation before we serialize.
+    runSILMandatoryOptPreDiffPasses(*sil_module);
```

Comment:
Does it work to run all the sil diagnostic passes before we serialize? That would be more consistent with what the normal compiler does in `FrontendTool.cpp : performCompileStepsPostSILGen()`

Reply:
Unfortunately, that messes up auto-diff. Specifically, mandatory inlining converts the high-level operators like `.+` to builtins, which causes autodiff to not compute correct gradients.

Reply:
Oh, I see that the comment above the `if` explains it.

I guess that the reason we don't encounter the same problems when using the normal compiler is that we never try to differentiate functions defined in different modules when using the normal compiler? Any time we do cross-module differentiation in the normal compiler, we require that the differentiated function is marked `@differentiable` so that it gets differentiated when the module gets compiled.